src/bfx_tools/vaxrank.ml

```diff
@@ -0,0 +1,180 @@
+*)
+module Configuration = struct
+  type t = {
+    name: string;
+    (* vaxrank-specific ones *)
+    vaccine_peptide_length: int;
+    padding_around_mutation: int;
+    max_vaccine_peptides_per_mutation: int;
+    max_mutations_in_report: int;
+    (* isovar-like ones *)
+    min_mapping_quality: int;
+    min_reads_supporting_variant_sequence: int;
+    use_duplicate_reads: bool;
+    drop_secondary_alignments: bool;
+    (* topiary-like ones *)
+    mhc_epitope_lengths: int list;
+    (* the rest *) 
+    parameters: (string * string) list;
+  }
+  let to_json {
+    name; 
+    vaccine_peptide_length; 
+    padding_around_mutation; 
+    max_vaccine_peptides_per_mutation; 
+    max_mutations_in_report; 
+    min_mapping_quality;
+    min_reads_supporting_variant_sequence;
+    use_duplicate_reads;
+    drop_secondary_alignments;
+    mhc_epitope_lengths;
+    parameters}: Yojson.Basic.json 
+    =
+    `Assoc [
+      "name", `String name;
+      "vaccine_peptide_length", `Int vaccine_peptide_length;
+      "padding_around_mutation", `Int padding_around_mutation;
+      "max_vaccine_peptides_per_mutation", `Int max_vaccine_peptides_per_mutation;
+      "max_mutations_in_report", `Int max_mutations_in_report;
+      "min_mapping_quality", `Int min_mapping_quality;
+      "min_reads_supporting_variant_sequence", 
+        `Int min_reads_supporting_variant_sequence;
+      "use_duplicate_reads", `Bool use_duplicate_reads;
+      "drop_secondary_alignments", `Bool drop_secondary_alignments;
+      "mhc_epitope_lengths", 
+        `List (List.map mhc_epitope_lengths ~f:(fun i -> `Int i));
+      "parameters",
+      `Assoc (List.map parameters ~f:(fun (a, b) -> a, `String b));
+    ]
+  let render {
+    name; 
+    vaccine_peptide_length; 
+    padding_around_mutation; 
+    max_vaccine_peptides_per_mutation; 
+    max_mutations_in_report; 
+    min_mapping_quality;
+    min_reads_supporting_variant_sequence;
+    use_duplicate_reads;
+    drop_secondary_alignments;
+    mhc_epitope_lengths;
+    parameters} 
+    =
+    sprintf "--vaccine-peptide-length %d" vaccine_peptide_length ::
+    sprintf "--padding-around-mutation %d" padding_around_mutation ::
+    sprintf "--max-vaccine-peptides-per-mutation %d" 
+      max_vaccine_peptides_per_mutation ::
+    sprintf "--max-mutations-in-report %d" max_mutations_in_report ::
+    sprintf "--min-mapping-quality %d" min_mapping_quality ::
+    sprintf "--min-reads-supporting-variant-sequence %d" 
+      min_reads_supporting_variant_sequence ::
+    (if use_duplicate_reads
+      then "--use-duplicate-reads" else "") ::
+    (if drop_secondary_alignments
+      then "--drop_secondary_alignments" else "") ::
+    sprintf "--mhc_epitope_lengths %s"
+      (mhc_epitope_lengths
+        |> List.map ~f:string_of_int
+        |> String.concat ~sep:",") ::
+    (List.concat_map parameters ~f:(fun (a,b) -> [a; b]))
+    |> List.filter ~f:(fun s -> not (String.is_empty s))
```

Comment:
Yes so here you are contenating the arguments with space like for a "shell command" call, but in the `~make` program you use `exec`.